homework/assignment_4/hw4.py

```diff
@@ -0,0 +1,216 @@
+            decision_tree,
+            feature_names=feature_names,
+            class_names=class_names,
+            out_file=dot_data,
+            filled=True,
+        )
+        graph = pydot.graph_from_dot_data(dot_data.getvalue())
+        graph[0].write_pdf(file_out + ".pdf")  # must access graph's first element
+        graph[0].write_png(file_out + ".png")  # must access graph's first element
+
+
+# first pass testing on titanic
+def main():
+    df_titanic = pd.read_csv("./datasets/titanic.csv")
+    df_titanic = df_titanic.dropna()
+
+    # split for titanic
+    column_to_move = df_titanic.pop("Survived")  # remove survived - dependent variable
+    df_titanic[
+        "Survived"
+    ] = column_to_move  # add survived (DV) to end of df to split easier
+
+    # removing useless columns
+    df_titanic.pop("Ticket")
+    df_titanic.pop("Cabin")
+    df_titanic.pop("PassengerId")
+    df_titanic.pop("Name")
+
+    X = df_titanic.iloc[:, :-1]
+    y = df_titanic.iloc[:, -1]
+
+    # categorizing each independent (predictor) variable
+    print_heading("Predictors")
+    catg = pd.DataFrame()
+    cont = pd.DataFrame()
+    for (columnName, columnData) in X.iteritems():
+        print("Column Name : ", columnName)
+        print("Unique Values : ", columnData.nunique())
+        if columnData.nunique() <= 3:
+            catg[columnName] = columnData
+            print("Boolean\n")
+        elif columnData.dtype == "object" and columnData.nunique() <= 3:
+            catg[columnName] = columnData
+            print("Boolean\n")
+        else:
+            cont[columnName] = columnData
+            print("Continuous\n")
+
+    # checking catg/cont
+    print(catg.head())
+    print(cont.head())
+
+    # categorizing dependent variable
+    print_heading("Dependent Variable")
+    print("Column Name : ", y.name)
+    print("Unique Values : ", y.nunique())
+    if y.nunique() > 2:  # we just need to check for 2
+        print("Continuous")
+    else:
+        print("Boolean")
+
+    # Linear model for continuous features
+    headers = list(cont.columns)
+
+    print(headers)
+    print("columns: ", cont.columns[0])
+    for key in cont:
+        titanic_features = key
+        col = cont[key]
+        predictor = sm.add_constant(col)
+        lm = sm.OLS(y, predictor)
+        lm_fitted = lm.fit()
+        print(f"Variable: {titanic_features}")
+        print(lm_fitted.summary())
+
+        # Get the p/t stats
+        t_value = round(lm_fitted.tvalues[1], 6)
+        p_value = "{:.6e}".format(lm_fitted.pvalues[1])
+
+        # Plot the figure
+        fig = px.scatter(x=y, y=col, trendline="ols")
+        fig.update_layout(
+            title=f"Variable: {titanic_features}: (t-value={t_value}) (p-value={p_value})",
+            xaxis_title=f"y: {y.name}",
+            yaxis_title=f"Variable: {titanic_features}",
+        )
+        fig.show()
+
+    # checking relationship between variables
+    print(X.corr())
```

Comment:
Might want to make statement longer with more description so we know which variables you're talking about.